br/pkg/task/stream.go

```diff
@@ -892,17 +897,48 @@ func RunStreamRestore(
 		ctx = opentracing.ContextWithSpan(ctx, span1)
 	}
 
-	StreamStorage := cfg.Config.Storage
+	logStartTS, truncateTS, err := getLogStartTS(ctx, cfg)
+	if err != nil {
+		return errors.Trace(err)
+	}
 
 	if len(cfg.FullBackupStorage) > 0 {
-		cfg.Config.Storage = cfg.FullBackupStorage
-		if err := RunRestore(ctx, g, FullRestoreCmd, cfg); err != nil {
+		cfg.StartTS, err = getFullBakcupTS(ctx, cfg)
+		if err != nil {
 			return errors.Trace(err)
 		}
+		if cfg.StartTS < logStartTS {
+			return errors.Annotatef(berrors.ErrInvalidArgument,
+				"it has gap between full backup ts:%d(%s) and log backup ts:%d(%s)"+
+					"you can \"restore full\" and \"restore point\" separately",
+				cfg.StartTS, oracle.GetTimeFromTS(cfg.StartTS),
+				logStartTS, oracle.GetTimeFromTS(logStartTS))
+		}
 	}
 
-	cfg.Config.Storage = StreamStorage
-	if err := restoreStream(ctx, g, cmdName, cfg); err != nil {
+	if logStartTS < truncateTS {
+		logStartTS = truncateTS
+	}
```

Comment:
should we do that before we check the relationship between `cfg.StartTS` and `logStartTS`?